linux/just_files/docker_functions.bsh

```diff
@@ -754,6 +793,125 @@ function docker_compose_volumes()
   done
 }
 
+#**
+# :Arguments: * ``$1`` - Name of service of interest
+#             * [``$2``] - Prefix to volume names
+# :Input: **stdin** - Yarped docker compose yaml file
+# :Output: * ``DOCKER_VOLUME_SOURCES`` - List of sources
+#          * ``DOCKER_VOLUME_TARGETS`` - Corresponding list of targets
+#          * ``DOCKER_VOLUME_FLAGS`` - Corresponding list of flags. Can be a mix of short and long format flags. Needs more parsing before this mix is useful.
+#          * ``DOCKER_VOLUME_FORMATS`` - Corresponding list specifying either "long" or "short" as the format of each volume
+#
+# Get volume lists from ``DOCKER_VOLUME_LINES``
+#
+# .. rubric:: Usage
+#
+# .. code-block:: bash
+#
+#    vsi::docker::get_compose_volumes nb < <(docker compose config | yarp)
+#    docker_compose_volumes
+#
+# .. seealso::
+#    yarp
+#**
+function vsi::docker::get_compose_volumes()
+{
+  local volume_line
+  vsi_compose_volume_sources=()
+  vsi_compose_volume_targets=()
+  vsi_compose_volume_flags=()
+  vsi_compose_volume_types=()
+  local -i count=0
+
+  local short_flag
+
+  local pattern="services\\.${1}\\.volumes\\[([0-9]+)](\\.[^ ]*)? = (.*)"
```

Comment:
could you give an example of this pattern. it would help in understanding how "" means the short format